src/task.ts

```diff
@@ -451,3 +412,34 @@ export const getSendTaskMatrixResult = (
     }
   }
 }
+
+export const cancelTask = async (ctx: Context, taskId: Task | string) => {
+  const {
+    taskDb: { db },
+  } = ctx
+
+  const task =
+    typeof taskId === "string"
+      ? await (async () => {
+          try {
+            return JSON.parse(await db.get(taskId)) as Task
+          } catch (error) {
+            if (error instanceof LevelErrors.NotFoundError) {
+              return error
+            } else {
+              throw error
+            }
+          }
+        })()
+      : taskId
+
+  if (task instanceof Error) {
+    return task
+  }
```

Comment:
I'm against using `let` (mutability) for convenience because `task` doesn't need to change after this point.

Reply:
I strongly disagree with you on this one, as this hardly brings any type safety, nor saves you from mutating `task` later, but makes the code considerably more complex.

Although, I do understand that this could be an issue of high "holywarness", so do the change or resolve the conversation at your discretion.